client/state/notification-settings/reducer.js

```diff
@@ -0,0 +1,70 @@
+/** @format */
+/**
+ * Internal dependencies
+ */
+import { combineReducers } from 'state/utils';
+import toggleState from 'state/notification-settings/toggle-state';
+import {
+	NOTIFICATION_SETTINGS_FETCH,
+	NOTIFICATION_SETTINGS_FETCH_COMPLETE,
+	NOTIFICATION_SETTINGS_FETCH_FAILED,
+	NOTIFICATION_SETTINGS_TOGGLE_SETTING,
+	NOTIFICATION_SETTINGS_SAVE,
+	NOTIFICATION_SETTINGS_SAVE_COMPLETE,
+	NOTIFICATION_SETTINGS_SAVE_FAILED,
+} from 'state/action-types';
+
+function toggleSetting( state, source ) {
+	if ( toggleState[ source ] ) {
+		return toggleState[ source ].apply( this, arguments );
+	}
+
+	return toggleState.blog.apply( this, arguments );
```

Comment:
The `toggleState.blog` function should be renamed to `toggleState.blogs` and then we can always use the `toggleState[ source ]` version.

Reply:
I definitely want to get rid of the need for this intermediate `toggleSetting` step and instead have the code be more explicitly defined.

Reply:
It turns out that this is somewhat necessary, or at least convenient... Sometimes the `source` is a number (blog id). I'm going to leave this as is for now just because I'm not sure how far reaching any changes would be :/

Reply:
Oh, now I see the code that passes `blogId` as `source`, too. Makes sense :+1: